### PR TITLE
[#2260] Check if adapter enabled for tenant earlier on

### LINKED ISSUE #2260: Do "adapter enabled for tenant" check before credentials validation
Followup of #2115:
Currently, a connection attempt from a device means that the target protocol adapter does some tenant related checks only *after* the credentials given by the device have been validated.

As the credentials validation may be quite expensive, it would be better to do the tenant related checks *before* that already, as soon as tenant information has been obtained from the credentials.

The tenant related checks are:
- check whether  the adapter is enabled for the tenant,
- check whether the connection limit for the tenant is not exceeded,
- check whether the message limit for the tenant is not exceeded by the request payload (HTTP/CoAP adapter).

EDIT: only the check whether the adapter is enabled for the tenant is to be done here - see below.

### PR DESCRIPTION
**Pull request by calohmn**
This fixes #2260.

### CODE REVIEW

File: service-base/src/main/java/org/eclipse/hono/service/http/AuthHandlerTools.java [hunk 1]
```diff
@@ -40,13 +40,19 @@ private AuthHandlerTools() {
...
+     * In all other cases, the routing context is failed with the given exception.
+     * <p>
+     * Note that the routing context is failed with just the exception, no status
+     * code. Setting the status code on the response corresponding to the exception
+     * is to be done in the failure handler, e.g. as implemented in the {@link DefaultFailureHandler}.
      *
      * @param ctx The routing context.
      * @param exception The cause of failure to process the request.
      * @param authenticateHeader The value to return in the HTTP Authenticate header.
+     * @see io.vertx.ext.web.handler.impl.AuthHandlerImpl#processException(RoutingContext, Throwable)
```
Comment (kaniyan) [inline service-base/src/main/java/org/eclipse/hono/service/http/AuthHandlerTools.java:55]: `AuthHandlerImpl#processException` is inaccessible from this class as that method is _protected_.

File: adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java [hunk 2]
```diff
@@ -331,6 +332,7 @@ protected Router createRouter() {
      * information before the potentially expensive credentials validation is done
      * <p>
      * The default implementation updates the trace sampling priority in the execution context tracing span.
+     * It also verifies that the adapter is enabled for the tenant, failing the returned future if that is not the case.
```
Comment (kaniyan) [inline adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java:335]: In addition to the adapter, `isAdapterEnabled(...)` also verifies if the tenant is _enabled_ or not. How about including this info about tenant also in the JavaDoc ?  It is also applicable for the MQTT adpter too. WDYT?

**Review by kaniyan — CHANGES_REQUESTED** [review]

File: service-base/src/main/java/org/eclipse/hono/service/http/AuthHandlerTools.java (hunk 1 above)
Comment (calohmn) [reply service-base/src/main/java/org/eclipse/hono/service/http/AuthHandlerTools.java:55]: There was a discussion on such a case earlier already (regarding `TableAdapterStore` class) and the outcome was, that the javadoc tool won't reject this with an error anyway, so it was deemed ok.

File: adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java (hunk 2 above)
Comment (calohmn) [reply adapters/http-vertx-base/src/main/java/org/eclipse/hono/adapter/http/AbstractVertxBasedHttpProtocolAdapter.java]: I've adapted the javadocs.

**Review by kaniyan — APPROVED** [review]
LGTM

**Review by sophokles73 — APPROVED** [review]
LGTM

**Pull request merged**